forch/mab_state_machine.py

```diff
@@ -0,0 +1,159 @@
+"""Module that implements MAB state machine"""
```

Comment:
state machine is not specific to MAB... it's the device authentication status. MAB is just one means of authentication, but everything else applies to other techniques.

Reply:
But the state machine is MAB. The states and transitions are specific.

Reply:
The only aspect of it that's MAB specific is the radius request and associated trigger. This entire structure would have value, for example, if we wanted to do something based off of IP address. The core of the state machine is managing the radius authenticated lifecycle, right?  The way it's named, if there was another system that needed to do IP-based authentication, or EAPoL-based authentication it would be another class entirely, which I don't think should be necessary.

None of the actual states involved are specific to MAB:

    SENT_REQ = "Sent MAB request"
    RECV_ACCPT = "Received RADIUS accept"
    RECV_REJ = "Received RADIUS reject"
    REQ_TIMEOUT = "RADIUS request timeout"
    AUTH_TIMEOUT = "Auth session timed out"
    EXPIRE = "Host expired"